migration.md

```diff
@@ -1,5 +1,57 @@
 # Migration
 
+## 0.5.0
+
+### Package exports
+
+[Information about Package entry points](https://nodejs.org/api/packages.html#package-entry-points)
+
+We now have a package exports entry in our `package.json`.
+This is to add better support for SSR which uses Node.js module resolution rather than the browser's algorithm.
+One example is that in Node.js a path like `foo/bar` doesn't automatically resolve to `foo/bar/index.js`.
+
+Another big advantage of package exports is pinning down explicitly where consumers can import from, increasing reliability and making explicit our public and private APIs. We can now change our file locations or edit filenames without being worried that some consumer was relying on it through an import we did not intend.
+
+For named imports, use:
+
+```js
+import { Foo } from '@divriots/starter-simba';
+```
+
+For named imports from specific simba packages, use:
+
+```js
+import { Bar } from '@divriots/starter-simba/input';
+```
+
+For custom elements definitions, use:
+
+```js
+import '@divriots/starter-simba/input/define';
+```
+
+When a package exports multiple custom elements, the `define` entrypoint will define all of them. 
+In case you only need a specific one, for example a `checkbox` but not the `checkbox-group` or `checkbox-indeterminate`, use:
+
+```js
+import '@divriots/starter-simba/checkbox/define-checkbox';
+```
+
+> Directly importing private API will no longer be supported if you use tools that correctly respect package `exports`, hence the breaking change:
+
+```js
+import { NotYours } from '@divriots/starter-simba/private/index.js'; // no longer possible in tools respecting our package.json exports entry
+```
+
+### Fixed dependencies on @lion
+
+Dependencies on `@lion` are now fixed instead of using a caret. This is due to the fact that Lion also internally fixes dependencies on itself.
+Not doing this could result in multiple undedupable installations of things like `@lion/core` or `@lion/form-core` resulting in fatal errors.
+The reason the installations would be undedupable because e.g. simba would rely on `^0.5.3` of `@lion/form-core` which would resolve to for example `0.5.5`, whereas
+internally lion would also rely on this package but with a fixed `0.5.3` meaning NPM would not dedupe the `0.5.3` installation to the `0.5.5` installation.
```

Comment:
I never liked that policy of Lion to pin dependencies and this is why
I think the pros do not overweight the cons, so Lion should fix that seeing how consumers run into this problem
IMO pinned deps should go to the 2nd tier if needed (so simba or ING's internal tier can do it if needed)
calling @daKmoR for help on this if he sees notifications

Reply:
I too, forgot why exactly we had to pin versions, I vaguely remember the main reason being that every install of let's say simba 0.5.0 should install the same version of lion button no matter what. If I don't pin versions in simba, people with the same version of simba as me could get different outcomes. Not pinning lion's internal dependencies causes the exact same issue except a layer deeper which is why I think we decided to pin. So yes, pinning versions solves an issue. However:

- Breaking changes are done in minors (beta) or majors (>1.0.0), so even if you don't get the exact same install of lion for the same version of simba over time, you should not get different outcomes to a point where this inconveniences you, you should only get non-breaking fixes/features. Pinning versions solves a symptom, but the cause is (accidentally) doing a breaking release without a breaking semver bump.
- We now see that undedupable installations are indeed an issue due to fixed versions, every mismatch of version cannot be deduped even though they might be semver compatible, the lack of caret prevents deduping here. This means that subclassers of lion are forced to also pin their lion dependencies on fixed versions, counter-intuitive, and it means you and your users don't get semver compatible changes like bugfixes for free anymore upon new installs :(
- It also means that importing directly from @lion as a Simba user is inherently dangerous, as you will need to manually ensure that whatever version of @lion you rely on directly is fixed to the exact same version that simba relies on, otherwise you get undedupable installs again.

Given these cons, I totally agree with @bashmish that I don't think pinning lion dependencies is worth it at all, the only thing that would convince me otherwise is if we need pinning for our monorepo release orchestration in @lion but I really doubt that's a requirement for the monorepo to work properly tbh.